File: leetcode/rust-leetcode/src/2095.delete-the-middle-node-of-a-linked-list.rs
// Definition for singly-linked list.
// #[derive(PartialEq, Eq, Clone, Debug)]
// pub struct ListNode {
//   pub val: i32,
//   pub next: Option<Box<ListNode>>
// }
// 
// impl ListNode {
//   #[inline]
//   fn new(val: i32) -> Self {
//     ListNode {
//       next: None,
//       val
//     }
//   }
// }
impl Solution {
    pub fn delete_middle(mut head: Option<Box<ListNode>>) -> Option<Box<ListNode>> {
        let mut fast = head.clone().unwrap();
        if fast.next.is_none() {
            return None;
        }
        fast = fast.next.unwrap();
        let mut slow = head.as_mut().unwrap();
        while fast.next.is_some() && fast.next.as_ref().unwrap().next.is_some() {
            fast = fast.next.unwrap().next.unwrap();
            slow = slow.next.as_mut().unwrap();
        }
        slow.next = slow.next.clone().unwrap().next;
        head
    }
}

File: leetcode/python/2095.delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None or head.next is None:
            return None
        fast = head.next
        slow = head
        while fast.next is not None and fast.next.next is not None:
            slow = slow.next
            fast = fast.next.next
        
        slow.next = slow.next.next
        
        return head

